### DISCUSSION TITLE
Most useful changes relative to vanilla Vim?

### DISCUSSION THREAD

**Discussion by leungbk** [macvim-dev/macvim#1238] [General]
I'm inexperienced with both Mac and Vim. What are the most impactful changes that Macvim provides?

I am aware that Macvim provides numerous Cmd-based bindings for commands like `:cnext`, but I have not been using them since I have other bindings for them already. I like Macvim's menu bar and full-screen capability.

**Comment by eirnym** [comment]
MacVim is a GUI for macOS. 
It provides some level of integration without breaking the experience. 
It adds some commands to modify menu and touch bar if you have it. Also icon is conform the Apple guidelines and look-and-feel for an OS. The main goal is to support the latest macOS version retaining some level of compatibility for older versions. 

Default key bindings (with Cmd prefix) are sane for a macOS user, the rest is untouched.

**Comment by ychin** [comment]
I really need to make this clearer, and put this into the README (which is… still really bare after all these years. I need to get on top of it) so it stays updated. If you do `:h macvim` it also gives a good sense (see https://github.com/macvim-dev/macvim/blob/master/runtime/doc/gui_mac.txt).

The basic thing that MacVim provides is a GUI for Vim on the Mac, while trying to hook in to macOS features. There isn't a vanilla up-to-date implementation of gVim on macOS, so MacVim is your best option if you want gVim on Mac (say if you want scroll bars or menus). MacVim also doesn't try to be more than a gVIm implementation, so if you want say an IDE that significantly goes beyond Vim, then MacVim is probably not for you.

Some features that MacVim provides outside of standard gVim features include:

1. Trackpad support. You can map swipe gestures and force touch to Vim commands.
2. Touch Bar support. You can add any touch bar functionality and associate them with any commands you want (using `:menu TouchBar.HelloWorld :echo 'Hello!'`).
3. Full-screen. MacVim supports two modes: a custom one, and the other one using macOS full screen.
4. Cmd key mapping. You can't map cmd-key combos easily otherwise in terminal Vim.

Some other features that I'm working on include:

1. Make Option-key mapping "just works™". Currently there is a big hack setting called `macmeta`, which lets you map keys like `<M-a>`, but it prevents you from being able to use Option key to enter texts the macOS way (e.g. using Option-; to enter "…"). This is how terminals handle it as well. 
2. Overlay scrollbars and provide one per Vim window.
3. Hardware accelerated renderer, and smoother scrolling along with some other features.
